homeassistant/components/mqtt/client.py

```diff
@@ -517,7 +584,7 @@ async def async_subscribe(
         # Only subscribe if currently connected.
         if self.connected:
             self._last_subscribe = time.time()
-            await self._async_perform_subscriptions(((topic, qos),))
+            await self._async_queue_subscriptions(((topic, qos),))
```

Comment:
As discussed on discord, maybe we should only use the debouncer during the DISCOVERY_COOLDOWN period?

Reply:
That would work for a retained setup and when reloading. If if after cooldown some service like Zigbee2MQTT sets up, then the issue remains.

Reply:
Adjusted to have a longer period during discovery cooldown, as discussed.

Reply:
> As discussed on discord, maybe we should only use the debouncer during the DISCOVERY_COOLDOWN period?

Suggestion is to do this an a separate PR after this PR is merged, so we can reuse the code.

Reply:
You still need the debouncer, even after the discovery phase.

If I add a new device to Zigbee2MQTT, it's going to cause n new entities to be created.  They could (and probably will) all share the same topic.

Additionally, if I re-start zigbee2mqtt, it will re-send all the discovery topics and statuses for everything.  I'm not sure if this would kick you back into the 'discovery' mode (as the MQTT module itself will have been started up for a long time).  Especially if it's re-sending all the discovery topics you already discovered on (and as such, really don't need to process).

In other words, it's safer to always debounce, but doing so with a longer time period during discovery makes sense.

Reply:
> You still need the debouncer, even after the discovery phase.

We will still have the debouncer after discovery cooldown, but with a smaller time.

Reply:
Yeah, I noticed - I just saw you mentioning removing it after discovery in another PR, and I was suggesting to not do that.